src/tribler-core/tribler_core/modules/metadata_store/restapi/metadata_endpoint.py

```diff
@@ -147,7 +147,8 @@ def setup_routes(self):
 
             :statuscode 404: if the torrent is not found in the database
         """
-        timeout = request.query.get('timeout', 20)
+        timeout = request.query.get('timeout')
+        timeout = int(timeout) if timeout and timeout.isdigit() else 20
```

Comment:
Please, don't just silently ignore wrong input because it will hide usage errors in the future. Instead, make the endpoint produce an error (something like "Wrong request"), see our tests. Also, produce a test for it. It is easy to do it by copy-pasting the existing tests.

Reply:
done.

Reply:
which also makes me think, the test should also fail here because the test already provide timeout value in the query.

https://github.com/Tribler/tribler/blob/09fbd720a6c251ffa76bfb2d2560b60a6920994d/src/tribler-core/tribler_core/modules/metadata_store/restapi/tests/test_metadata_endpoint.py#L231

was this test even passing? I am confused :S

Reply:
Ah i think i found the root cause,

https://github.com/Tribler/tribler/blob/09fbd720a6c251ffa76bfb2d2560b60a6920994d/src/tribler-core/tribler_core/modules/torrent_checker/torrent_checker.py#L302

The returned res obejct should be iterated and exception types should not be returned as succesfull results. Otherwise all extentions in the coro will be examined as successful, this is why the test scripts didnt work in this case. @ichorid

Reply:
Well, TorrentChecker was always kind of a lost child... No one owns the code and no one wants to touch it without good reason. So, can you fix it? :wink:

Reply:
Sure no problem, but where to redirect the exceptions and their tracebacks? Is logging.error ok?

Reply:
I checked the code of torrent checker, there are some ugly stuff there, feels like i need to pull the forest itself to pull the banana.

I suggest to limit this pr rest end point limited. I will do a final check of the test script, i dont think it will work yet but it will cover the scenario of the changes. I will request a review when done hopefully in the afternoon.

Reply:
Exceptions should be raised as normal to trigger a bug report. However, incorrect input into REST endpoint should never raise an exception but instead, send an "error" response and log the error as a warning.